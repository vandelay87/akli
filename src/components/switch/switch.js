import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { color } from '../../styles/colors'
import { robotoRegular } from '../../styles/fonts'

const Switch = ({ id, checked, handleToggle, label }) => (
  <>
    <StyledSwitch
      id={id}
      type="checkbox"
      role="switch"
      checked={checked}
      onChange={handleToggle}
    />
    <StyledLabel htmlFor={id}>
      <StyledLabel.Text>{label}</StyledLabel.Text>
      <StyledThumb role="presentation" />
    </StyledLabel>
  </>
)

const StyledSwitch = styled('input')`
  visibility: hidden;
  width: 0;
  height: 0;

  &:checked + label {
    background: ${color.primary};

    span:nth-child(2) {
      transform: translateX(12px);
    }
  }
`
const StyledLabel = styled('label')`
  display: flex;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  width: 32px;
  height: 14px;
  background: #808080;
  border-radius: 7px;
  position: relative;
  transition: background-color 0.2s;
`
StyledLabel.Text = styled('span')`
  ${robotoRegular}
  margin: 0 auto 0 40px;
  text-align: center;
`
const StyledThumb = styled('span')`
  content: '';
  position: absolute;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  background: ${color.onPrimary};
  box-shadow: 0 0 2px 0 rgba(10, 10, 10, 0.29);
  transition: transform 0.2s;
`

Switch.propTypes = {
  id: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  handleToggle: PropTypes.func.isRequired,
  label: PropTypes.string,
}
Switch.defaultProps = {
  label: '',
}

export default Switch
