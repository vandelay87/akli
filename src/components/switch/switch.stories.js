import React from 'react'
import { text, boolean } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'
import Switch from './switch'

export default {
  component: Switch,
  title: 'Switch',
}

export const switchWithLabel = () => {
  const checkedKnob = boolean('Checked', true)
  const labelKnob = text('Label', 'dark mode')

  return (
    <Switch
      id="switch"
      checked={checkedKnob}
      handleToggle={action('switch-toggle')}
      label={labelKnob}
    />
  )
}
