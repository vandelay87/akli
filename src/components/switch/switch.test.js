import renderer from 'react-test-renderer'
import { switchWithLabel } from './switch.stories'

describe('switch.js', () => {
  it('renders component', () => {
    const switchTree = renderer.create(switchWithLabel()).toJSON()

    expect(switchTree).toMatchSnapshot()
  })
})
