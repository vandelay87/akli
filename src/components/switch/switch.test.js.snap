// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch.js renders component 1`] = `
Array [
  <input
    checked={true}
    className="sc-bdVaJa jTNZgx"
    id="switch"
    onChange={[Function]}
    role="switch"
    type="checkbox"
  />,
  <label
    className="sc-bwzfXH eQCjbJ"
    htmlFor="switch"
  >
    <span
      className="sc-htpNat hIwdvW"
    >
      dark mode
    </span>
    <span
      className="sc-bxivhb aNRpm"
      role="presentation"
    />
  </label>,
]
`;
